### PR TITLE
[MKL-DNN] Design document on acquire API

### PR DESCRIPTION
**Pull request by jczaja**
This PR is providing document on design and usage (for developers of paddle mkl-dnn kernels) of Acquire API.

Rendered version:
https://github.com/jczaja/FluidDoc/blob/prv-acquire/doc/fluid/design/mkldnn/acquire_api/acquire_api.md

@grygielski, @Sand3r-, @bingyanghuang, @luotao1  Please review

### CODE REVIEW

File: doc/fluid/design/mkldnn/acquire_api/acquire_api.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+#### 5. Registering MKL-DNN memory format in corresponding Tensor
+Last step is to register mkldnn output memory object format inside of Output tensor eg. set Tensor::format_ to mkl-dnn enum that corresponds the way Tensor data is arranged (NCHW, NCHW16C etc.) This enum can be taken from dst memory object (wrapper to Output tensor) in Forward pass or from diff_src memory object (wrapper to X_grad Tensor). 
+
+Example of registring MKL-DNN format in output tensor:
+
+    auto output_format =
+        (mkldnn::memory::format)dst_memory->get_primitive_desc()
+            .desc()
+            .data.format;
```
Comment (grygielski) [inline doc/fluid/design/mkldnn/acquire_api/acquire_api.md]: Please change it to `auto output_format = platform::GetMKLDNNFormat(*dst_memory);` as this is the function doing exactly same thing, but it's much more clear and won't be outdated once mkl-dnn v1.0 come.

**Review by grygielski — CHANGES_REQUESTED** [review]
Great work, really good explanation! I would suggest just one change I mentioned below.

File: doc/fluid/design/mkldnn/acquire_api/acquire_api.md (hunk 1 above)
Comment (jczaja) [reply doc/fluid/design/mkldnn/acquire_api/acquire_api.md]: @grygielski good suggestion. I have introduced requested modification

**Review by grygielski — APPROVED** [review]
Thanks for introducing suggested changes! LGTM

File: doc/fluid/design/mkldnn/acquire_api/acquire_api.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Design Doc: MKL-DNN Acquire API
+
+MKL-DNN kernels that are using mkl-dnn API tend to be quite complex due to:
```
Comment (bingyanghuang) [inline doc/fluid/design/mkldnn/acquire_api/acquire_api.md]: mkl-dnn --> MKL-DNN

File: doc/fluid/design/mkldnn/acquire_api/acquire_api.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+* number of mkl-dnn api calls needed, which in fact are mostly repeated across all  mkl-dnn kernels
```
Comment (bingyanghuang) [inline doc/fluid/design/mkldnn/acquire_api/acquire_api.md]: mkl-dnn --> MKL-DNN Please unify the MKL-DNN in the doc.

File: doc/fluid/design/mkldnn/acquire_api/acquire_api.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+* caching mechanism of mkl-dnn objects  (similar concept in every paddle operator mkl-dnn kernel)
```
Comment (bingyanghuang) [inline doc/fluid/design/mkldnn/acquire_api/acquire_api.md]: kernel --> kernels

**Comment by jczaja** [comment]
@luotao1 Could you please review this PR?

File: doc/fluid/design/mkldnn/acquire_api/acquire_api.md (hunk 4 above)
Comment (jczaja) [reply doc/fluid/design/mkldnn/acquire_api/acquire_api.md]: ok

File: doc/fluid/design/mkldnn/acquire_api/acquire_api.md (hunk 3 above)
Comment (jczaja) [reply doc/fluid/design/mkldnn/acquire_api/acquire_api.md]: ok

File: doc/fluid/design/mkldnn/acquire_api/acquire_api.md (hunk 2 above)
Comment (jczaja) [reply doc/fluid/design/mkldnn/acquire_api/acquire_api.md]: ok

**Comment by jczaja** [comment]
@bingyanghuang Thanks for reviewing my changes. I have followed your suggestions.

**Comment by bingyanghuang** [comment]
@jczaja LGTM. @luotao1 Please start your review if you have the time.

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**